cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -117,12 +117,16 @@ private String ensureIndexed(Index index) {
         try {
             if(index.indexName == null){
                 // generate a name for the index.
-                String indexName = this.generateIndexName();
+                String indexName = this.generateIndexName(listIndexesInDatabaseQueue());
```

Comment:
Also, this is a duplicate database call -- there's another call to `listIndexesInDatabaseQueue()` just below.